src/common/datatypes/Geography.cpp

```diff
@@ -0,0 +1,164 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "common/datatypes/Geography.h"
+
+#include <folly/String.h>
+#include <folly/hash/Hash.h>
+#include <s2/s2loop.h>
+#include <s2/s2polygon.h>
+
+#include <cstdint>
+
+#include "common/geo/io/wkb/WKBReader.h"
+#include "common/geo/io/wkt/WKTWriter.h"
+
+namespace nebula {
+
+GeoShape Geography::shape() const {
+  // TODO(jie) May store the shapetype as the data member of Geography is ok.
+  std::string wkbCopy = wkb;
```

Comment:
why need to copy? this function has been const.

Reply:
The wkb data member is const because the function is const, and the class WKBReader's function will modify the pointer which points to the wkb bytes when reading it.
```
  StatusOr<std::unique_ptr<Geometry>> read(uint8_t *&beg, uint8_t *end) const;
```

Reply:
you can use to a pointer pointing to the `wkb.data()` and change the `pointer` in read function.

Reply:
ok, has fixed